File: javalin-openapi-processor/src/main/java/net/dzikoysk/openapi/processor/annotations/OpenApiSecurityInstance.java
package net.dzikoysk.openapi.processor.annotations;

import io.javalin.plugin.openapi.annotations.OpenApiSecurity;
import net.dzikoysk.openapi.processor.processing.AnnotationMirrorMapper;

import javax.lang.model.element.AnnotationMirror;
import javax.lang.model.element.AnnotationValue;
import java.lang.annotation.Annotation;

@SuppressWarnings("ClassExplicitlyAnnotation")
final class OpenApiSecurityInstance extends AnnotationMirrorMapper implements OpenApiSecurity {

    public OpenApiSecurityInstance(AnnotationMirror mirror) {
        super(mirror);
    }

    @Override
    public String name() {
        return getString("name");
    }

    @Override
    public String[] scopes() {
        return getArray("scopes", AnnotationValue.class, value -> value.getValue().toString(), String[]::new);
    }

    @Override
    public Class<? extends Annotation> annotationType() {
        return OpenApiSecurity.class;
    }

}

File: javalin-openapi-processor/src/main/kotlin/net/dzikoysk/openapi/processor/annotations/OpenApiSecurityInstance.kt
package net.dzikoysk.openapi.processor.annotations

import io.javalin.plugin.openapi.annotations.OpenApiSecurity
import net.dzikoysk.openapi.processor.processing.AnnotationMirrorMapper
import javax.lang.model.element.AnnotationMirror

class OpenApiSecurityInstance(mirror: AnnotationMirror) : AnnotationMirrorMapper(mirror) {

    fun name(): String =
        getString("name")

    fun scopes(): List<String> =
        getArray("scopes", String::class.java)

    fun annotationType(): Class<out Annotation> =
        OpenApiSecurity::class.java

}